### ISSUE TITLE
Testing advice page

### ISSUE THREAD

**Issue by pherjung** [sailfishos/docs.sailfishos.org#70]
As discussed at the community meeting from 14th April, I create this issue to discuss specifically about the content of this page.
I first though to a page listing all functions and expected behaviour, but it will be a huge work and I think the testing team already do this.
As mentioned on Community Meeting, best is an test advice page but I don't know exactly which contain belongs there.

**Comment by pherjung** [comment]
Is it right that new features should be tested first as well as changes? So, based on the release note it is possible to know what to test?

**Comment by jpwalden** [comment]
> Is it right that new features should be tested first as well as changes? So, based on the release note it is possible to know what to test?

Yes, this is a good approach and way to check which are the most relevant areas to be tested since when some area has changes those changes might have caused regression.

**Comment by deloptes** [comment]
I spent several years in orchestrating and managing test cycles.  IMO the test goals should come in collaboration with developers/managers. The test scenarios should reflect the impact of the changes and should include regression tests
As we (users) do not know what is being changed exactly, one can not know what exactly to test.
This means that the Release Notes should be made available in kind of beta, so that users can do the testing in EA phase.

**Comment by pherjung** [comment]
> This means that the Release Notes should be made available in kind of beta

I don't understand what you mean. Once a new update is available as EA, a release note is already present too.

**Comment by deloptes** [comment]
> > This means that the Release Notes should be made available in kind of beta
> 
> I don't understand what you mean. Once a new update is available as EA, a release note is already present too.

Well, if they are available in EA phase, then one should also plan and communicate the tests.
Again, you as user can not do this unless you know what was changed and what was impacted.
But I can imagine that there are some test cases that users can perform. You can start with a list in specific domains: people app, phone call app etc. Define a template for the test cases and define test cases. usually test cases are derived from use cases. So you need a list of valid use cases. Development and management should be definitely involved here.
While developers write unit tests, I found out they are not the same as UAT.

**Comment by jpwalden** [comment]
1st version created here: https://docs.sailfishos.org/Develop/Platform/Testing_Advice/
all the feedback and comments are welcome so that we can gather needed content there.
